src/arguments.rs

```diff
@@ -241,6 +244,20 @@ pub fn parse_arguments() -> ArgMatches<'static> {
             Ok(_) => Ok(()),
         });
 
+    let key_pair_algorithm = Arg::with_name(&Parameters::algo.as_ref())
+        .long(&Parameters::algo.as_ref())
+        .takes_value(true)
+        .required(false)
+        .help("Algorithm used to generate key pair. [Default: ECDSA or from context]")
+        .possible_value(trust::SignAlgo::ECDSA.as_ref())
+        .possible_value(trust::SignAlgo::EdDSA.as_ref());
+
+    let algo_arg = Arg::with_name(&Parameters::algo.as_ref())
+        .required(true)
+        .help("Algorithm used to generate key pair.")
+        .possible_value(trust::SignAlgo::ECDSA.as_ref())
+        .possible_value(trust::SignAlgo::EdDSA.as_ref());
```

Comment:
if the only différence is that one is optional and the other required then you could use only one, and `clone()` it to change the required state. 
See my comment in the context subcommand

Reply:
The problem with this is we have to specify `.long()` in one and not in another. But no issues I have made this change